src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -49,19 +55,34 @@ public Map<String, User> groupByEmail(final List<User> users) {
      * @return map {age : people with this age}
      */
     public Map<Integer, List<Person>> groupByAge(final List<Person> people) {
-        return null;
+        Map<Integer, List<Person>> userMap = new HashMap<>();
+        people.forEach(person -> {
+            List<Person> peopleThisAge = people.stream()
+                    .filter(person1 -> Objects.equals(person1.getAge(), person.getAge()))
+                    .collect(Collectors.toList());
+            userMap.put(person.getAge(), peopleThisAge);
+        });
```

Comment:
There's simpler way to achieve this kind of grouping. Please, try to find it.
Refer to the presentation slides, you might find a hint there.